### PR TITLE
gh-122431: Disallow negative values in `readline.append_history_file`

### PR DESCRIPTION
**Pull request by ZeroIntensity**
* Issue: gh-122431

FWIW, this is actually a bug upstream in GNU's `append_history` function, but this is the best that Python can do in the meantime to prevent accessing out of bounds memory.

### CODE REVIEW

**Comment by ZeroIntensity** [comment]
I'm quite confused on how the documentation tests could be related.

File: Lib/test/test_readline.py [hunk 1]
```diff
@@ -113,6 +113,14 @@ def test_write_read_append(self):
         # write_history_file can create the target
         readline.write_history_file(hfilename)
 
+        # Negative values should be disallowed
+        with self.assertRaises(ValueError):
+            readline.append_history_file(-42, hfilename)
+
+        # See gh-122431, using the minumum signed integer value caused a segfault
```
Comment (vstinner) [inline Lib/test/test_readline.py]:
```suggestion
        # See gh-122431, using the minimum signed integer value caused a segfault
```

File: Modules/readline.c [hunk 2]
```diff
@@ -351,6 +351,12 @@ readline_append_history_file_impl(PyObject *module, int nelements,
                                   PyObject *filename_obj)
 /*[clinic end generated code: output=5df06fc9da56e4e4 input=784b774db3a4b7c5]*/
 {
+    if (nelements < 0)
+    {
+        PyErr_SetString(PyExc_ValueError, "nelements cannot be negative");
```
Comment (vstinner) [inline Modules/readline.c]:
```suggestion
        PyErr_SetString(PyExc_ValueError, "nelements must be positive");
```

**Review by vstinner — APPROVED** [review]
LGTM, thanks for the update.

**Pull request merged**